### PR TITLE
[Dev]- Bug[12725] - Fixed date picker value when the date value is cleared.

### PR DESCRIPTION
**Pull request by bittu1028**

### CODE REVIEW

File: src/Modules/DatePicker.react.js [hunk 1]
```diff
@@ -386,7 +386,11 @@ class DatePicker extends Component {
                 this.props.onChange(dateObj);
             }
         } else {
-            this.props.onChange(date.unix());
+            if (date) {
```
Comment (groberts314) [inline src/Modules/DatePicker.react.js]: is `if (moment.isMoment(date))` maybe slightly safer here?
https://momentjs.com/docs/#/query/is-a-moment/

Being a mostly C# dude I tend to be leary of simple truthiness or falsiness checks and lean towards more precise checks (using the various helpers in lodash and such, which I _love_).

**Comment by morethanfire** [comment]
@groberts314 any updates on this PR?

**Comment by groberts314** [comment]
@morethanfire None from me.  If this looks good to you, let's :shipit: 
Have you reviewed?

**Comment by morethanfire** [comment]
What release are we thinking this will be?

**Comment by groberts314** [comment]
It is likely needed for the 1.14.0 HC Back Office release.  @bittu1028 would have to advise to be sure, but that's my best guess.

**Comment by groberts314** [comment]
Or did you mean what React CM UI version should we assign this?

**Comment by groberts314** [comment]
If we need/want to do a release just for this, it can be 1.18.4 (next patch level version number), in my opinion.  You wanna do it, or shall I?

**Comment by morethanfire** [comment]
I guess I am kind of asking for both. I don’t want give HC a version that causes regression testing.

Sent from my iPhone

On Feb 19, 2019, at 10:36 AM, Geoffrey Roberts <notifications@github.com<mailto:notifications@github.com>> wrote:


If we need/want to do a release just for this, it can be 1.18.4 (next patch level version number), in my opinion. You wanna do it, or shall I?

**Comment by groberts314** [comment]
Some amount of that is unavoidable :-)  Can't make an omelette without breaking some eggs! 🥚 🍳

**Pull request merged**